pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1945,8 +1945,12 @@ public boolean getBoolean(int columnIndex) throws SQLException {
       return false; // SQL NULL
     }
 
+    int col = columnIndex - 1;
+    if (Oid.BOOL == fields[col].getOID()) {
+      return 't' == (char) (this_row[col][0] & 0xFF);
```

Comment:
Index bounds are checked first in this function.
Should be faster, as it does not allocate new objects.
Just for my understanding: is it necessary to check TEXT_FORMAT/BINARY_FORMAT?